cmd/manager/main.go

```diff
@@ -98,6 +100,19 @@ func main() {
 	go registrySyncer.Sync(stopCh)
 	go catalogsourceconfig.Syncer.Sync(stopCh)
 
+	client, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		exit(err)
+	}
+	upgrader := migrator.NewUpgrader(log.NewEntry(log.New()), client)
```

Comment:
These seem like implementation details that should be obfuscated by the migrator. Why can't we just do all of this with one call? We should just do something like:

```
upgrader := migrator.NewUpgrader(log.NewEntry(log.New()), client)
err := upgrader.Migrate()
if err != nil {
	exit(err)
}
```

And you should implement a method on that migrator type that updates subscriptions and cleans up resources as implementation details.

If we start adding the details of _how_ we do the migration, especially if there are a lot of steps in future iterations, it will be really hard to read `main.go` -- which to me is already too long and complex.

Reply:
As an aside, this will also make it easier to write unit tests for the migrator package.